src/python_testing/matter_testing_support.py

```diff
@@ -268,6 +270,37 @@ async def read_single_attribute(self, dev_ctrl: ChipDeviceCtrl, node_id: int, en
         data = result[endpoint]
         return list(data.values())[0][attribute]
 
+    async def read_single_attribute_check_success(self, cluster: object, attribute: object, dev_ctrl: ChipDeviceCtrl=None, node_id: int=None, endpoint: int=0) -> object:
+        if dev_ctrl is None:
+            dev_ctrl = self.default_controller
+        if node_id is None:
+            node_id = self.dut_node_id
+
+        result = await dev_ctrl.ReadAttribute(node_id, [(endpoint, attribute)])
+        attr_ret = result[endpoint][cluster][attribute]
+        err_msg = "Error reading {}:{}".format(str(cluster), str(attribute))
+        asserts.assert_true(attr_ret is not None, err_msg)
+        asserts.assert_false(isinstance(attr_ret, Clusters.Attribute.ValueDecodeFailure), err_msg)
+        return attr_ret
+
+    async def read_single_attribute_expect_error(self, cluster: object, attribute: object, error: Status, dev_ctrl: ChipDeviceCtrl=None, node_id: int=None, endpoint: int=0) -> object:
+        if dev_ctrl is None:
+            dev_ctrl = self.default_controller
+        if node_id is None:
+            node_id = self.dut_node_id
+
+        result = await dev_ctrl.ReadAttribute(node_id, [(endpoint, attribute)])
+        attr_ret = result[endpoint][cluster][attribute]
+        err_msg = "Did not see expected error when reading {}:{}".format(str(cluster), str(attribute))
+        asserts.assert_true(attr_ret is not None, err_msg)
+        asserts.assert_true(isinstance(attr_ret, Clusters.Attribute.ValueDecodeFailure), err_msg)
+        asserts.assert_true(isinstance(attr_ret.Reason, InteractionModelError), err_msg)
+        asserts.assert_equal(attr_ret.Reason.status, error, err_msg)
+        return attr_ret
+
+    def print_step(self, stepnum: int, title:str) -> None:
```

Comment:
Spacing, looks odd to have both styles in same line.

Reply:
Turns out my auto-formatter was off. New commit fixes this.